addons/purchase/controllers/portal.py

```diff
@@ -118,7 +177,11 @@ def portal_my_purchase_order(self, order_id=None, access_token=None, **kw):
         if confirm_type == 'reception':
             order_sudo._confirm_reception_mail()
 
-        values = self._purchase_order_get_page_view_values(order_sudo, access_token, **kw)
+        if order_sudo.state in ('sent'):
+            history = 'my_rfqs_history'
+        else:
+            history = 'my_purchases_history'
+        values = self._purchase_order_get_page_view_values(order_sudo, access_token, history, **kw)
```

Comment:
This part could be in _purchase_order_get_page_view_values and don't create an extra argument on the method